### PR TITLE
Abba value set support

### PR DESCRIPTION
**Pull request by jbradl11**
An initial attempt to add ValueSet support to the Module Builder. Any place that supports a code or multiple codes *should* now support a Value set.

### CODE REVIEW

**Review by mgramigna — CHANGES_REQUESTED** [review]
The way we have defined this in the GMF, we are going with all lowercase `valueset`. So we would expect the outputted format from the module builder to call the attribute `valueset`. Not unifying this leads to two key side effects:

1. The module is not exported in a way that the backend is expecting
2. Pasting JSON into the module builder with an existing `valueset` attribute bugs out

**Review by mgramigna — CHANGES_REQUESTED** [review]
Some tests are failing because it is rendering a component that now depends on the new prop you introduces, so you'll also need to update the `State` test to use a mock function for `renderCodeOrValueSet`. I would just follow what they do for the other functions they mock for these tests

File: src/components/editor/State.js [hunk 1]
```diff
@@ -2363,6 +2383,27 @@ class Observation extends Component<Props> {
...
+           <a className='editable-text' onClick={() => {this.props.onChange('value_code')({val: {id: null}}); this.props.onChange('value_code')({val: {id: getTemplate('Type.Code.Snomed')}}); }}>Add Code</a> ValueSet
+           <br />
+           <ValueSet valueSet={state.value_code} onChange={this.props.onChange('value_code')} />
+         </div>
+       );
+     }
+   }
+
   renderToggles(currentItem) {
     console.log(currentItem);
```
Comment (mgramigna) [inline src/components/editor/State.js]: It doesn't look like you actually added this??? Either way let's get this outta here ;)

**Review by mgramigna — CHANGES_REQUESTED** [review]
Somewhere along the way with your recent changes, we lost the display or `url` and `display` in the actual state's box on the right side when a valueset is entered.

image

**Review by mgramigna — CHANGES_REQUESTED** [review]
:/

So I went through all of the state types and these are the problems I found. Fortunately I think they're all related, but I'm still sorry.

`AllergyEnd`, `CarePlanEnd`, `ConditionEnd`, `Death`: Clicking "Add ValueSet" causes the box to disappear. Did we decide if we wanted to support adding a valueset for "End" states? I think we did, but the backend isn't ready for it yet?? Easiest solution might be to just remote the button from the end states.

`MedicationEnd`: Clicking "Add Codes" does nothing

**Review by mgramigna — APPROVED** [review]
Don't let this task taint your view of React

**Review by hossenlopp — APPROVED** [review]
I'm gonna approve this now. I think we will cross the bridge in regard to the code vs valueset for attributes in the GMF parsing in Synthea proper. May have to revisit it if it's an issue.

**Pull request merged**